services/app-api/utils/validation/validation.ts

```diff
@@ -2,6 +2,22 @@ import * as yup from "yup";
 import { nested, endDate, schemaMap } from "./schemaMap";
 import { AnyObject } from "../types/types";
 import { error } from "../constants/constants";
+import { APIGatewayEvent } from "aws-lambda";
+
+// validate report keys
+export const validReport = (
+  event: APIGatewayEvent,
+  apiMethod: string | undefined
+) => {
+  const pathParameters = event.pathParameters!;
+  if (apiMethod === "create") {
+    return pathParameters.reportType && pathParameters.state;
+  } else {
+    return (
+      pathParameters.reportType && pathParameters.state && pathParameters.id
+    );
```

Comment:
Nit: A quick note on this pattern. No need to do `else {}` when using an `if → return` pattern, because if it got caught in the `if`, it will already have returned, so you can just return without the `else`, like:

```js
const boolChecker = () => {
  if (whatever) {
    return true;
  }
  return false;
}
```